packages/core/src/types.ts

```diff
@@ -0,0 +1,58 @@
+declare global {
+  interface GlobalContext {
+    __merkur__: Merkur;
+  }
+}
+
+export type AnyFn = (...args: unknown[]) => unknown;
+export type AnyObj = Record<string, unknown>;
+// export type Widget = AnyObj; // TODO
+
+export type RegisterFunction = ({
+  name,
+  version,
+  createWidget,
+}: {
+  name: string;
+  version: string;
+  createWidget: CreateFunction;
+}) => void;
+
+export type Plugin = {
+  setup: (merkur: Merkur) => void;
```

Comment:
Plugin has 'create' and 'setup' method with 'widget' object and must return `widget` object.

Reply:
Both methods are optional